bundles/org.openhab.binding.myuplink/src/main/java/org/openhab/binding/myuplink/internal/Utils.java

```diff
@@ -0,0 +1,256 @@
+     * get element as BigDecimal.
+     *
+     * @param jsonObject
+     * @param key
+     * @return
+     */
+    public static @Nullable BigDecimal getAsBigDecimal(@Nullable JsonObject jsonObject, String key) {
+        JsonElement element = jsonObject == null ? null : jsonObject.get(key);
+        return (element == null || element instanceof JsonNull) ? null : element.getAsBigDecimal();
+    }
+
+    /**
+     * get element as boolean.
+     *
+     * @param jsonObject
+     * @param key
+     * @return
+     */
+    public static @Nullable Boolean getAsBool(@Nullable JsonObject jsonObject, String key) {
+        JsonElement json = jsonObject == null ? null : jsonObject.get(key);
+        return (json == null || json instanceof JsonNull) ? null : json.getAsBoolean();
+    }
+
+    /**
+     * null safe version of getAsBool with default value.
+     *
+     * @param jsonObject
+     * @param key
+     * @return
+     */
+    public static boolean getAsBool(@Nullable JsonObject jsonObject, String key, Boolean defaultValue) {
+        Boolean result = getAsBool(jsonObject, key);
+        return result == null ? defaultValue : result;
+    }
+
+    /**
+     * retrieves typeID of a channel.
+     *
+     * @param channel
+     * @return typeID or empty string if typeUID is null.
+     */
+    public static String getChannelTypeId(Channel channel) {
+        ChannelTypeUID typeUID = channel.getChannelTypeUID();
+        if (typeUID == null) {
+            return "";
+        }
+        return typeUID.getId();
+    }
+
+    /**
+     * retrieves the validation expression which is assigned to this channel, fallback to a public static, if no
+     * validation
+     * is
+     * defined.
+     *
+     * @param channel
+     * @return the validation expression
+     */
+    public static String getValidationExpression(Channel channel) {
+        String expr = getPropertyOrParameter(channel, PARAMETER_NAME_VALIDATION_REGEXP);
+        if (expr == null) {
+            throw new ConfigurationException(
+                    "channel (" + channel.getUID().getId() + ") does not have a validation expression configured");
+        }
+        return expr;
+    }
+
+    /**
+     * internal utiliy method which returns a property (if found) or a config parameter (if found) otherwise null
+     *
+     * @param channel
+     * @param name
+     * @return
+     */
+    public static @Nullable String getPropertyOrParameter(Channel channel, String name) {
+        String value = channel.getProperties().get(name);
+        // also eclipse says this cannot be null, it definitely can!
+        if (value == null || value.isEmpty()) {
+            Object obj = channel.getConfiguration().get(name);
+            value = obj == null ? null : obj.toString();
+        }
+        return value;
+    }
+
+    /**
+     * capitalize method derived from WordUtils.
+     *
+     * @param str
+     * @return
+     */
+    public static String capitalize(final String str) {
```

Comment:
Quickly scanned the code and it seems the core api can be used here:
https://www.openhab.org/javadoc/latest/org/openhab/core/util/stringutils#capitalize(java.lang.String)

Reply:
fixed